SU2_CFD/src/transfer_structure.cpp

```diff
@@ -137,46 +144,76 @@ void CTransfer::Scatter_InterfaceData(CSolver *donor_solution, CSolver *target_s
     for (iMarkerDonor = 0; iMarkerDonor < nMarkerDonor; iMarkerDonor++) {
       /*--- If the tag GetMarker_All_FSIinterface(iMarkerDonor) equals the index we are looping at ---*/
       if ( donor_config->GetMarker_All_FSIinterface(iMarkerDonor) == iMarkerInt ) {
-        /*--- We have identified the local index of the Donor marker ---*/
-        /*--- Store the number of local points that belong to Marker_Donor on each processor ---*/
-        /*--- This are the number of points that will be sent from this particular processor ---*/
-        /*--- This includes the halo nodes ---*/
-        nLocalVertexDonor = donor_geometry->GetnVertex(iMarkerDonor);
-        /*--- Store the identifier for the structural marker ---*/
         Marker_Donor = iMarkerDonor;
         /*--- Exit the for loop: we have found the local index for iMarkerFSI on the FEA side ---*/
         break;
       }
-      else {
-        /*--- If the tag hasn't matched any tag within the donor markers ---*/
-        nLocalVertexDonor = 0;
-        Marker_Donor = -1;
-      }
     }
     
     /*--- On the target side ---*/
     
     for (iMarkerTarget = 0; iMarkerTarget < nMarkerTarget; iMarkerTarget++) {
       /*--- If the tag GetMarker_All_FSIinterface(iMarkerFlow) equals the index we are looping at ---*/
       if ( target_config->GetMarker_All_FSIinterface(iMarkerTarget) == iMarkerInt ) {
-        /*--- We have identified the local index of the Flow marker ---*/
-        /*--- Store the number of local points that belong to Marker_Flow on each processor ---*/
-        /*--- This includes the halo nodes ---*/
-        nLocalVertexTarget = target_geometry->GetnVertex(iMarkerTarget);
-        /*--- Store the identifier for the fluid marker ---*/
         Marker_Target = iMarkerTarget;
         /*--- Exit the for loop: we have found the local index for iMarkerFSI on the FEA side ---*/
         break;
       }
-      else {
-        /*--- If the tag hasn't matched any tag within the Flow markers ---*/
-        nLocalVertexTarget = 0;
-        Marker_Target = -1;
+    }
+
+    #ifdef HAVE_MPI
+
+    Donor_check  = -1;
+    Target_check = -1;
+
+    /*--- We gather a vector in MASTER_NODE that determines if the boundary is not on the processor because of the partition or because the zone does not include it  ---*/
```

Comment:
Yeah, the previous version of the code could deal with 2 zone problems only.
In that loop the code tries to find the marker corresponding to the fsi_interface so that 2 zones can communicate.
Those lines I added are needed to understand whether the marker does not match the interface due to the fact that the two zones are not actually sharing the interface or because of the partition due to parallel execution. In the latter case the rank which owns the marker broadcast the information to all processes so that they know that those 2 zones needs to communicate.